src/audio/base_fw.c

```diff
@@ -525,6 +525,52 @@ static int basefw_get_large_config(struct comp_dev *dev,
 						data_offset, data);
 };
 
+/**
+ * Handles the DMA Control IPC message to initialize or modify DMA gateway configuration.
+ *
+ * @param first_block Indicates if this is the first data block in the message.
+ * @param last_block Indicates if this is the last data block in the message.
+ * @param data_offset The offset of the data in the message.
+ * @param data Pointer to the data buffer containing the DMA Control message.
+ * @return 0 on success, error code on failure.
+ */
+static int basefw_dma_control(bool first_block,
+			      bool last_block,
+			      uint32_t data_offset,
+			      const char *data)
+{
+	struct ipc4_dma_control *dma_control;
+	size_t data_size;
+	int ret;
+
+	/* Ensure that the message is atomic and contains all necessary information */
+	if (!first_block || !last_block) {
+		tr_err(&ipc_tr, "Non-atomic DMA Control message received");
+		return IPC4_ERROR_INVALID_PARAM;
+	}
+
+	dma_control = (struct ipc4_dma_control *)data;
+	if (dma_control->config_length > 0) {
+		/* DMA Control is passed using a structure with the same construction as in DAI
+		 * configuration. There is an additional section whose size is not accounted for in
+		 * the config_length field. As a result, the configuration size will always be 0.
+		 */
+		tr_err(&ipc_tr, "The expected size of the data is 0.");
+		return IPC4_ERROR_INVALID_PARAM;
+	}
+
+	data_size = data_offset - (sizeof(struct ipc4_dma_control) - sizeof(uint32_t));
+	ret = basefw_vendor_dma_control(dma_control->node_id,
+					(const char *)dma_control->config_data,
```

Comment:
I would rather pass `dma_control->config_data`  and `dma_control->config_length` as is here. Tlv data struct, which is typically used in `config_data` is based on DW, so It will be easier to handle it later.

Reply:
As I already wrote above, `config_length` has a value of 0 in the case of SSP and such logic will not work there.